rdwatch/core/views/region.py

```diff
@@ -1,14 +1,174 @@
+import json
+import tempfile
+
 from ninja import Router
 from ninja.pagination import PageNumberPagination, paginate
+from ninja.responses import Response
 
-from django.http import HttpRequest
+from django.db import connection
+from django.db.models import BooleanField, F, Field, Func, Q
+from django.http import HttpRequest, HttpResponse, JsonResponse
+from django.shortcuts import get_object_or_404
 
-from rdwatch.core.models import Region
+from rdwatch.core.models import ModelRun, Region
+from rdwatch.core.schemas import RegionModel
 
 router = Router()
 
 
+# indicates why a region can't be deleted by the current User
+def get_delete_block(region_id, user):
+    region = Region.objects.get(pk=region_id)
+    if region.owner is None and region.public:
+        return 'Region is a System Region that is Public'
+    if region.owner != user:
+        return 'You do not have permission to delete this region.'
+    if ModelRun.objects.filter(region=region_id).exists():
+        return (
+            'This region is currently being used in model runs and cannot be deleted.'
+        )
+    return False
+
+
 @router.get('/', response=list[str])
 @paginate(PageNumberPagination, page_size=1000)
 def list_regions(request: HttpRequest):
     return Region.objects.values_list('name', flat=True)
+
+
+@router.get('/details', response=list[dict])
+@paginate(PageNumberPagination, page_size=1000)
+def list_region_details(request: HttpRequest):
+    regions = Region.objects.values_list(
+        'name', 'owner__username', 'public', 'pk', 'geom'
+    )
+    return [
+        {
+            'name': region[0],
+            'owner': region[1] if region[1] is not None else 'None',
+            'value': region[0] if region[1] is None else f'{region[0]}_{region[1]}',
+            'public': region[2],
+            'id': region[3],
+            'deleteBlock': get_delete_block(region[3], request.user),
+            'hasGeom': True if region[4] else False,
+        }
+        for region in regions
+    ]
+
+
+@router.post('/')
+def post_region_model_editor(
+    request: HttpRequest,
+    region_model: RegionModel,
+):
+    owner = request.user
+    # Extract the owner from the request user
+    region, _created = Region.create_region_model_from_geoJSON(
+        region_model, True, owner
+    )
+
+    return Response(
+        {'detail': 'Region model created successfully', 'region_id': str(region.name)},
+        status=201,
+    )
```

Comment:
It should be possible to simplify this to
```suggestion
    return 201, {'detail': 'Region model created successfully', 'region_id': str(region.name)}
```
If we want to return structured output that has a `detail` and `region_id` field, I think this should be defined as a `Schema` instead of using a raw `Reponse` object.